datafusion/optimizer/src/optimizer.rs

```diff
@@ -250,10 +250,6 @@ impl Optimizer {
             Arc::new(DecorrelatePredicateSubquery::new()),
             Arc::new(ScalarSubqueryToJoin::new()),
             Arc::new(ExtractEquijoinPredicate::new()),
-            // simplify expressions does not simplify expressions in subqueries, so we
```

Comment:
❤️ 

This may also make planning non trivially faster as SimplifyExpressions is quite expensive